src/server/handlers/openapi/mojaloop/parties/{Type}/{ID}.ts

```diff
@@ -30,37 +31,63 @@ import { PartiesTypeIDPutRequest } from '~/shared/ml-thirdparty-client/models/op
 
 import { Status } from '~/models/transaction'
 import { transactionRepository } from '~/repositories/transaction'
+import { consentRepository } from '~/repositories/consent'
+import { ConsentStatus } from '~/models/consent'
 
 /**
  * Handles callback from Mojaloop that specifies detailed info about a requested party.
- * 
+ *
  * @param context   an object that contains detailed information about the incoming request.
  * @param request   original request object as defined by the hapi library.
  * @param h         original request toolkit as defined by the hapi libary.
  */
-export const put: Handler = async (context: Context, _: Request, h: ResponseToolkit): Promise<ResponseObject> => {
+export const put: Handler = async (
+  context: Context,
+  _: Request,
+  h: ResponseToolkit
+): Promise<ResponseObject> => {
   // Retrieve the data that have been validated by the openapi-backend library.
-  let body = context.request.body as PartiesTypeIDPutRequest
-  let partyIdType = context.request.params.Type
-  let partyIdentifier = context.request.params.ID
+  const body = context.request.body as PartiesTypeIDPutRequest
+  const partyIdType = context.request.params.Type
+  const partyIdentifier = context.request.params.ID
 
   // Find all matching documents in Firebase that are waiting for the result of
-  // party lookup with the specified type and identifier. The execution of this 
-  // function is expected to run asynchronously, so the server could quickly 
+  // party lookup with the specified type and identifier. The execution of this
+  // function is expected to run asynchronously, so the server could quickly
   // give a response to Mojaloop.
-  transactionRepository.update(
-    // Conditions for the documents that need to be updated
-    {
-      "payee.partyIdInfo.partyIdType": partyIdType,
-      "payee.partyIdInfo.partyIdentifier": partyIdentifier,
-      "status": Status.PENDING_PARTY_LOOKUP,
-    },
-    // Update the given field by their new values
-    {
-      payee: body.party,
-      status: Status.PENDING_PAYEE_CONFIRMATION,
-    }
-  )
+
+  if (partyIdType === 'OPAQUE') {
```

Comment:
Add a note why this is OK. (Opaque is during linking when you're fetching the accounts available for linking from a pre-determined DFSP).

Reply:
+1, I think it would be better to add some comments in the code that explains about the use of opaque identifiers and how it is special